Dockerfile

```diff
@@ -88,28 +90,36 @@ RUN apt-get update && apt-get install -y ruby
 # --create-home creates a home folder, ie /home/<username>
 RUN useradd --create-home agent
 USER agent
+WORKDIR /home/agent
 
 # install agent executable
-COPY --chown=agent:agent --from=build /root/agent/build/bin/agent /usr/local/bin/
+COPY --chown=agent:agent --from=build /root/agent/*.zip .
 
 # copy data to /etc/mtconnect
-COPY --chown=agent:agent --from=build /root/agent/schemas /etc/mtconnect/schemas
-COPY --chown=agent:agent --from=build /root/agent/simulator /etc/mtconnect/simulator
-COPY --chown=agent:agent --from=build /root/agent/styles /etc/mtconnect/styles
+RUN unzip *.zip
```

Comment:
I believe you need to install `unzip` first, as on Ubuntu 22.10 it is not installed by default.

Reply:
I tested it on 22.04 w/o a problem. Are you suggesting we switch to 22.10? It's not an LTS. I'm a bit confused.

Reply:
I believe you haven’t read my comments in #300 (I think GitHub email notifications does not include the collapsed code created using `<details><summary>...`).

In [`CMakeLists.txt` (L8-9)](https://github.com/mtconnect/cppagent/blob/main/CMakeLists.txt#L8-L9), you state that the agent requires `cmake@3.23+`. On Ubuntu 22.04 LTS, the latest version is `v3.22`, therefore I went ahead and updated the Ubuntu image to 22.10. Yes, it is a non-LTS image.

I have also tried the `ubuntu:22.04` image too, but there is no `unzip` installed on it either. Does Docker runs `unzip` without the need to install it first? :thinking:

```bash
# docker run --rm -it ubuntu:22.04 bash -c 'unzip'
bash: line 1: unzip: command not found
```

Reply:
I believe you haven’t read my comments in #300 (I think GitHub email notifications does not include the collapsed code created using `<details><summary>...`).

In [`CMakeLists.txt` (L8-9)](https://github.com/mtconnect/cppagent/blob/main/CMakeLists.txt#L8-L9), you state that the agent requires `cmake@3.23+`. On Ubuntu 22.04 LTS, the latest version is `v3.22`, therefore I went ahead and updated the Ubuntu image to 22.10. Yes, it is a non-LTS image.

I have also tried the `ubuntu:22.04` image too, but there is no `unzip` installed on it either. Does Docker runs `unzip` without the need to install it first? :thinking:

```bash
# docker run --rm -it ubuntu:22.04 bash -c 'unzip'
bash: line 1: unzip: command not found
```

Reply:
Probably missed that comment. conan has a `tool_requires` for `cmake/3.26.4`. This does not mean it needs to be installed on the system, it will download and build if necessary or use the conancenter binaries. There is an install capability we use to get the dependencies that was introduced in 3.23.

conan takes care of this, we do not need to change the ubuntu version.

Reply:
I'll add zip to the dockerfile apt install. Not sure why there should be a diff in the version.

Reply:
Added in the latest push.

Reply:
> I'll add zip to the dockerfile apt install.

You don’t to install `zip`, but `unzip`. :thinking:

That said, `apt install zip` actually installs `unzip`.

```bash
# apt install zip
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  unzip
The following NEW packages will be installed:
  unzip zip
0 upgraded, 2 newly installed, 0 to remove and 4 not upgraded.
Need to get 350 kB of archives.
After this operation, 929 kB of additional disk space will be used.
Do you want to continue? [Y/n] y
```

> Not sure why there should be a diff in the version.

I have no idea, I just followed the `cmake` requiments. I am happy to keep using `ubuntu:22.04` if building works with `cmake@3.22` too.

Reply:
conan supplies cmake 3.26.4 required by the build. It is not an issue.

Reply:
If that is the case, do we still need to install `cmake` in [`Dockerfile` (L43)](https://github.com/mtconnect/cppagent/blob/add_deploy_release_cd_workflow/Dockerfile#L43)? :thinking: